src/ca/resourceset.rs

```diff
@@ -0,0 +1,418 @@
+use std::convert::TryFrom;
+use std::fmt;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use routecore::asn::Asn;
+
+use crate::repository::resources::{AsResources, IpResources};
+use crate::repository::Cert;
+use crate::repository::{
+    resources::{AsBlock, AsBlocks, AsBlocksBuilder, Ipv4Blocks, Ipv6Blocks},
+    roa::RoaIpAddress,
+};
+
+//------------ ResourceSet ---------------------------------------------------
+
+/// A set of ASN, IPv4 and IPv6 resources.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct ResourceSet {
```

Comment:
Maybe this should live under `repository::resources`? It may be useful for other purposes as well – e.g. you might want to actually peel these out of a resource certificate.

Reply:
Moved it - pushed